buildenv/jenkins/jobs/infrastructure/signedOffByCheck

```diff
@@ -0,0 +1,100 @@
+ *******************************************************************************/
+
+def FAIL = false
+def WRONG_COMMIT = false
+def SRC_REPO = 'https://github.com/${ghprbGhRepository}.git'
+def HASHES = '###################################'
+def COMMITS = []
+def BAD_COMMITS = []
+
+timeout(time: 6, unit: 'HOURS') {
+    stage('SignedOffBy Check') {
+        node ('worker') {
+            timestamps {
+                try {
+                    checkout changelog: false, poll: false,
+                        scm: [$class: 'GitSCM',
+                            branches: [[name: ghprbActualCommit]],
+                            doGenerateSubmoduleConfigurations: false,
+                            extensions: [[$class: 'CloneOption',
+                                            depth: 0,
+                                            noTags: false,
+                                            reference: '/home/jenkins/openjdk_cache',
+                                            shallow: false]],
+                            submoduleCfg: [],
+                            userRemoteConfigs: [[refspec: "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*",
+                                                    url: SRC_REPO]]]
+
+                    COMMITS = sh(
+                         script : "git log origin/${ghprbTargetBranch}..HEAD",
+                         returnStdout: true
+                         ).trim()
+                    if (COMMITS == "") {
+                        echo "There are no commits to check for signoffby message"
+                    } else {
+                        def COMMIT_RAW = COMMITS.split("(?=commit [a-f0-9]{40}+\n(Merge: [a-f0-9 ]+)*\n*Author: .*\nDate:   .*+\n)");
+                        COMMIT_RAW.each { COMMIT ->
+                            if (COMMIT != "") {
+                                def splitCommit = COMMIT.split("\\r?\\n")
+                                def lastBlankLine = 0
+                                for (int i = splitCommit.size() - 1; i >= 0; i--) {
+                                    if ( splitCommit[i] ==~ /^\s*$/ ) {
+                                        lastBlankLine = i
+                                        break  
+                                    }                                                               
+                                }
+                                for (int j = splitCommit.size() - 1; j >= lastBlankLine; j-- ) {
+                                    if ( splitCommit[j] ==~ /\s*Signed\-off\-by\:\s+[^<]+\s+\<[^@]+@[^@]+\.[^@]+\>/ ) {
+                                        echo "The following commit appears to have a proper sign-off:\n${COMMIT}" 
+                                        WRONG_COMMIT = false
+                                        break             
+                                    } else {
+                                        WRONG_COMMIT = true
```

Comment:
Rather than setting this to `true` in an `else`, can we set it to `true` at the start of each iteration. that way it is assumed bad unless proven otherwise?

Reply:
fixed